charmtools/build/tactics.py

```diff
@@ -750,35 +751,44 @@ def combine(self, existing):
         self.previous = existing.previous + [existing]
         return self
 
-    def _add(self, pip, wheelhouse, *reqs):
+    def _add(self, wheelhouse, *reqs):
         with utils.tempdir(chdir=False) as temp_dir:
             # put in a temp dir first to ensure we track all of the files
-            utils.Process((pip, 'install',
-                           '--no-binary', ':all:',
-                           '-d', temp_dir) +
-                          reqs).exit_on_error()()
+            self._pip('download', '--no-binary', ':all:', '-d', temp_dir,
+                      *reqs)
             for wheel in temp_dir.files():
                 dest = wheelhouse / wheel.basename()
                 dest.remove_p()
                 wheel.move(wheelhouse)
                 self.tracked.append(dest)
 
+    def _pip(self, *args):
+        # have to use bash to call pip to activate the venv properly first
+        utils.Process(('bash', '-c', ' '.join(
+            ('.', self._venv / 'bin' / 'activate', ';', 'pip3') + args
+        ))).exit_on_error()()
+
     def __call__(self, venv=None):
-        create_venv = venv is None
-        venv = venv or path(tempfile.mkdtemp())
-        pip = venv / 'bin' / 'pip3'
+        create_venv = self._venv is None
+        self._venv = self._venv or path(tempfile.mkdtemp())
         wheelhouse = self.target.directory / 'wheelhouse'
         wheelhouse.mkdir_p()
         if create_venv:
             utils.Process(
-                ('virtualenv', '--python', 'python3', venv)).exit_on_error()()
-            utils.Process(
-                (pip, 'install', '-U', 'pip', 'wheel')).exit_on_error()()
+                ('virtualenv', '--python', 'python3', self._venv)
+            ).exit_on_error()()
+            self._pip('install', '-U',
+                      '"pip>=9.0.0,<10.0.0"',
```

Comment:
Version 1.x is required in the `install_requires` because that has to match the pre-installed system pip in order for the command to launch.  This is installing a newer version of pip into the venv, which is needed to be able to build the wheelhouse.